solutions/src/test/java/ru/sberbank/school/task10/ConcurrentFixedPoolTest.java

```diff
@@ -0,0 +1,142 @@
+
+
+import org.junit.jupiter.api.*;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+
+class ConcurrentFixedPoolTest {
+
+    @Test
+    @DisplayName("Тест на выброс исключения при исполнении задачи без запуска пула")
+    void withoutStart() {
+        ConcurrentFixedPool fixedPool = new ConcurrentFixedPool(3);
+
+        Assertions.assertThrows(IllegalStateException.class, () -> fixedPool.execute(() -> {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+        ));
+
+    }
+
+    @Test
+    @DisplayName("Тест на выброс исключения при повторном запуске пула")
+    void restart() {
+        ConcurrentFixedPool fixedPool = new ConcurrentFixedPool(3);
+        fixedPool.start();
+        Assertions.assertThrows(IllegalStateException.class, fixedPool::start);
+    }
+
+    @Test
+    @DisplayName("Тест на выброс исключения при колличестве потоков меньше 1")
+    void illegalArgument() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new ConcurrentFixedPool(0));
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new ConcurrentFixedPool(-1));
+    }
+
+    @Test
+    @DisplayName("Тест на проверку состояния пула при остановке")
+    void stopPool() {
+        ConcurrentFixedPool fixedPool = new ConcurrentFixedPool(3);
+        fixedPool.start();
+        fixedPool.stopNow();
+        Optional<Thread> first = Arrays.stream(fixedPool.getThreads()).filter(Objects::nonNull).findFirst();
+
+        Assertions.assertFalse(fixedPool.isExist());
+        Assertions.assertEquals(0, fixedPool.getTasksCount());
+        Assertions.assertFalse(first.isPresent());
+    }
+
+    @Test
+    @DisplayName("Тест на выброс исключения при исполнении задачи после остановки пула")
+    void withoutStartAfterStopping() {
+        ConcurrentFixedPool fixedPool = new ConcurrentFixedPool(3);
+        fixedPool.start();
+        fixedPool.stopNow();
+        Assertions.assertThrows(IllegalStateException.class, () -> fixedPool.execute(() -> {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+        ));
+    }
+
+    @Test
+    @DisplayName("Тест на устойчивость пула к исключениям в задачах")
+    void exceptionInRunnable() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(10);
+        ConcurrentFixedPool fixedPool = new ConcurrentFixedPool(3);
+        fixedPool.start();
+
+        for (int i = 0; i < 10; i++) {
+            fixedPool.execute(() -> {
+                latch.countDown();
+                System.out.println(5/0);
```

Comment:
WhitespaceAround: &#39;/&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1588708321>*

Reply:
!